src/Raven.Server/Commercial/SetupManager.cs

```diff
@@ -762,6 +763,20 @@ private static void LogErrorAndThrow(Action<IOperationProgress> onProgress, Setu
             }
         }
 
+        public static void AssertCanListenToPort(int port)
+        {
+            var ipProperties = IPGlobalProperties.GetIPGlobalProperties();
+            var ipEndPoints = ipProperties.GetActiveTcpListeners();
+
+            foreach (var activePort in ipEndPoints)
```

Comment:
What if this is the same port as we are currently using?

Reply:
Also, this is wrong. You are checking _all_ listeners, but we are bound to IP / port, not just port.
so I may be listening on `127.0.0.1:443` but it is fine to setup binding for `127.0.0.2:443`